### ISSUE TITLE
`CKV_AWS_19`: False positive when S3 server side encryption has a conditional

### ISSUE THREAD

**Issue by gi-el** [bridgecrewio/checkov#7269]
**Describe the issue**
A false positive `CKV_AWS_19: Ensure all data stored in the S3 bucket is securely encrypted at rest` is triggered when the in Terraform the encryption of an S3 bucket is configured through an `sse_algorithm` in a `aws_s3_bucket_server_side_encryption_configuration` associated with an `aws_s3_bucket` contains a conditional. This was introduced in 3.2.458 (3.2.457 works as expected).

**Examples**
We have the following in a module:

```
resource "aws_s3_bucket" "this" {
  bucket = var.name

  lifecycle {
    prevent_destroy = true
  }

  tags = {
    "Name" = var.name
  }
}

resource "aws_s3_bucket_server_side_encryption_configuration" "this" {
  bucket = aws_s3_bucket.this.id

  rule {
    apply_server_side_encryption_by_default {
      sse_algorithm = var.kms_key_arn == "" ? "AES256" : "aws:kms"

      kms_master_key_id = var.kms_key_arn == "" ? "" : var.kms_key_arn
    }

    bucket_key_enabled = var.bucket_key_enabled
  }
}

**Expected outcome:**
Checkov should not flag this configuration as a CKV_AWS_19 violation, since the encryption is always one of the required values.

```
**Version (please complete the following information):**
 - Checkov Version 3.2.458

**Comment by rwlodarczyk-xealth** [comment]
Is there any update on this?
